overlord/state/change.go

```diff
@@ -171,6 +173,7 @@ func (c *Change) UnmarshalJSON(data []byte) error {
 	c.status = unmarshalled.Status
 	c.data = unmarshalled.Data
 	c.taskIDs = unmarshalled.TaskIDs
+	c.ready = make(chan struct{})
```

Comment:
I think, to be consistent, we should close c.ready here if status is ready, something to test in the checkpointing tests